gapis/resolve/dependencygraph2/graph_visualization/graph_visualization.go

```diff
@@ -16,17 +16,208 @@ package graph_visualization
+				currentHierarchy.levelId[currentHierarchy.currentLevel] = currentHierarchy.currentId
+				currentHierarchy.currentId++
+				currentHierarchy.currentLevel++
+			}
+		}
+	}
+	return label
+}
+
+func getSubCommandLabel(cmdNode dependencygraph2.CmdNode) string {
+	label := ""
+	for i := 1; i < len(cmdNode.Index); i++ {
+		label += fmt.Sprintf("/%d", cmdNode.Index[i])
+	}
+	return label
+}
 
+func getSplittedLabelByChar(label *string, splitChar byte) []string {
+	splitLabel := []string{}
+	lastPosition := 0
+	for i := 0; i <= len(*label); i++ {
+		if i == len(*label) || (*label)[i] == splitChar {
+			splitLabel = append(splitLabel, (*label)[lastPosition:i])
+			lastPosition = i + 1
+		}
+	}
+	return splitLabel
+}
+func getMaxCommonPrefixBetweenSplitLabels(splitLabel1 *[]string, splitLabel2 *[]string) int {
+	size := len(*splitLabel1)
+	if len(*splitLabel2) < size {
+		size = len(*splitLabel2)
+	}
+	for i := 0; i < size; i++ {
+		if (*splitLabel1)[i] != (*splitLabel2)[i] {
+			return i
+		}
+	}
+	return size
+}
+
+func getMaxCommonPrefixBetweenLabels(label1, label2 string) int {
+	splitLabel1 := getSplittedLabelByChar(&label1, '/')
+	splitLabel2 := getSplittedLabelByChar(&label2, '/')
+	return getMaxCommonPrefixBetweenSplitLabels(&splitLabel1, &splitLabel2)
+}
+
+func createGraphFromDependencyGraph(dependencyGraph dependencygraph2.DependencyGraph) (*Graph, error) {
+
+	numberOfNodes := dependencyGraph.NumNodes()
+	graph := createGraph(0)
+	currentHierarchy := &Hierarchy{}
+	previousNode := &Node{}
+	for i := 0; i < numberOfNodes; i++ {
+		dependencyNode := dependencyGraph.GetNode(dependencygraph2.NodeID(i))
+		if cmdNode, ok := dependencyNode.(dependencygraph2.CmdNode); ok {
+			cmdNodeId := cmdNode.Index[0]
+			command := dependencyGraph.GetCommand(api.CmdID(cmdNodeId))
+			commandName := command.CmdName()
+			label := getCommandLabel(currentHierarchy, command)
+			label += fmt.Sprintf("%s%d", commandName, cmdNodeId)
+			label += getSubCommandLabel(cmdNode)
+			label += "\""
+			attributes := fmt.Sprintf("\"%v\"", command.CmdParams())
+
+			graph.addNodeByIdAndLabelAndNameAndAttributes(i, label, commandName, attributes)
+
+			node := graph.nodeIdToNode[i]
+			if _, ok1 := commandsInsideRenderScope[previousNode.name]; ok1 {
+				if _, ok2 := commandsInsideRenderScope[node.name]; ok2 {
+					if getMaxCommonPrefixBetweenLabels(previousNode.label, node.label) >= 2 {
+						graph.addEdgeBetweenNodes(node, previousNode)
+					}
+				}
+			}
+			if _, ok := commandsInsideRenderScope[node.name]; ok {
+				previousNode = node
+			}
+		}
+	}
+
+	addDependencyToGraph := func(source, sink dependencygraph2.NodeID) error {
+		idSource, idSink := int(source), int(sink)
+		if sourceNode, ok1 := graph.nodeIdToNode[idSource]; ok1 {
+			if sinkNode, ok2 := graph.nodeIdToNode[idSink]; ok2 {
+				_, ok1 = commandsInsideRenderScope[sourceNode.name]
+				_, ok2 = commandsInsideRenderScope[sinkNode.name]
+				if ok1 == false || ok2 == false {
```

Comment:
Yes this is intentional. One of the initital ideas was consider commands inside render scope forms a chain , and remove all edges between them, but in the next commits this is removed , because the order of commands could be affected ,which I think is more important.
I am thinking to remove this because could cause confusion, but also this is part of all process we have been working so far.